pkg/dependency/credhelper/cred_helper_binary.go

```diff
@@ -0,0 +1,189 @@
+	l          flog.Logger
+	fc         *config.Finch
+	user       string
+	hcfg       helperConfig
+}
+
+var _ dependency.Dependency = (*credhelperbin)(nil)
+
+func newCredHelperBinary(fp path.Finch, fs afero.Fs, cmdCreator command.Creator, l flog.Logger, fc *config.Finch,
+	user string, hcfg helperConfig,
+) *credhelperbin {
+	return &credhelperbin{
+		// TODO: consider replacing fp with only the strings that are used instead of the entire type
+		fp:         fp,
+		fs:         fs,
+		cmdCreator: cmdCreator,
+		l:          l,
+		fc:         fc,
+		user:       user,
+		hcfg:       hcfg,
+	}
+}
+
+func updateConfigFile(bin *credhelperbin) error {
+	cfgPath := fmt.Sprintf("%s%s", bin.hcfg.finchPath, "config.json")
+	binCfgName := bin.credHelperConfigName()
+	fileExists, err := afero.Exists(bin.fs, cfgPath)
+	if err != nil {
+		return err
+	}
+	if !fileExists {
+		file, err := bin.fs.Create(cfgPath)
+		if err != nil {
+			return err
+		}
+		JSONstr := fmt.Sprintf("{\"credsStore\":\"%s\"}", binCfgName)
+		JSON := []byte(JSONstr)
+		_, err = file.Write(JSON)
+		if err != nil {
+			return err
+		}
+	} else {
+		fileRead, err := bin.fs.Open(cfgPath)
+		if err != nil {
+			return err
+		}
+		bytes, err := afero.ReadAll(fileRead)
+		if err != nil {
+			return err
+		}
+		var cfg configfile.ConfigFile
+		err = json.Unmarshal(bytes, &cfg)
+		if err != nil {
+			return err
+		}
+		credsStore := cfg.CredentialsStore
+		defer fileRead.Close() //nolint:errcheck // closing the file
+		if strings.Compare(credsStore, binCfgName) != 0 {
+			file, err := bin.fs.OpenFile(cfgPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0o755)
+			if err != nil {
+				return err
+			}
+			cfg.CredentialsStore = binCfgName
+			finalCfgBytes, err := json.Marshal(&cfg)
+			if err != nil {
+				return err
+			}
+			_, err = file.Write(finalCfgBytes)
+			if err != nil {
+				return err
+			}
+			defer file.Close() //nolint:errcheck // closing the file before exit
+		}
+	}
+
+	return nil
+}
+
+func (bin *credhelperbin) credHelperConfigName() string {
+	return strings.ReplaceAll(bin.hcfg.binaryName, "docker-credential-", "")
+}
+
+func (bin *credhelperbin) fullInstallPath() string {
+	return fmt.Sprintf("%s%s", bin.hcfg.installFolder, bin.hcfg.binaryName)
+}
+
+func (bin *credhelperbin) Installed() bool {
+	dirExists, err := afero.DirExists(bin.fs, bin.hcfg.installFolder)
+	if err != nil {
+		bin.l.Errorf("failed to get status of credhelperbin directory: %v", err)
+		return false
+	}
+	if !dirExists {
+		return false
+	}
+	fileExists, err := afero.Exists(bin.fs, bin.fullInstallPath())
+	if err != nil {
+		bin.l.Errorf("failed to get status of credential helper binary: %v", err)
+		return false
+	}
+	if !fileExists {
+		return false
+	}
+	file, err := bin.fs.Open(bin.fullInstallPath())
+	if err != nil {
+		bin.l.Error(err)
+		return false
+	}
+	hash, err := digest.FromReader(file)
+	if err != nil {
+		bin.l.Error(err)
+		return false
+	}
+	defer file.Close() //nolint:errcheck // closing the file
```

Comment:
nit: move this up to before the call to `digest.FromReader`. Otherwise, if reading that hash results in error, the successfully opened file would not be closed.